src/shared/components/plots/BoxScatterPlot.tsx

```diff
@@ -272,46 +282,99 @@ export default class BoxScatterPlot<
     }
 
     @computed get bottomLegendHeight() {
-        //height of legend in case its on bottom
-        if (!this.props.legendData) {
+        if (
+            !this.props.legendData ||
+            !this.props.legendData.length ||
+            this.legendLocation !== 'bottom'
+        ) {
             return 0;
         } else {
-            const numRows = Math.ceil(
-                this.props.legendData.length / LEGEND_ITEMS_PER_ROW
+            return getBottomLegendHeight(
+                this.legendItemsPerRow,
+                this.props.legendData,
+                this.props.legendTitle
             );
-            return 23.7 * numRows;
         }
     }
 
+    @computed get maxLegendLabelWidth() {
+        if (this.props.legendData) {
+            return getMaxLegendLabelWidth(this.props.legendData);
+        }
+
+        return 0;
+    }
+
+    @computed get legendItemsPerRow() {
+        return getLegendItemsPerRow(
+            this.maxLegendLabelWidth,
+            this.svgWidth,
+            LEGEND_COLUMN_PADDING,
+            this.props.legendTitle
+        );
+    }
+
     private get legend() {
         if (this.props.legendData && this.props.legendData.length) {
             let legendData = this.props.legendData;
             if (this.legendLocation === 'bottom') {
                 // if legend is at bottom then flatten labels
                 legendData = legendData.map(x => {
-                    let name = x.name;
-                    if (Array.isArray(x.name)) {
+                    let { name, ...rest } = x;
```

Comment:
neat.  this assigns everything but x.name to an object called "rest"?

Reply:
yep, im not sure the technical name for this but it rules